modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectDefinitionDetails.tsx

```diff
@@ -26,14 +30,60 @@ export function RightSidebarObjectDefinitionDetails({
 	companyKeyValuePair,
 	siteKeyValuePair,
 }: RightSidebarObjectDefinitionDetailsProps) {
-	const store = useStore();
-	const {nodes} = store.getState();
+	const [{elements}] = useFolderContext();
 
-	const selectedNode = nodes.find(
-		(objectDefinitionNode: Node<ObjectDefinitionNodeData>) => {
-			return objectDefinitionNode.data?.nodeSelected;
+	const selectedNode = elements.find((element) => {
+		if (isNode(element)) {
+			return (element as Node<ObjectDefinitionNodeData>).data
+				?.nodeSelected;
 		}
-	);
+	}) as Node<ObjectDefinitionNodeData>;
+
+	const [
+		nonRelationshipObjectFieldsInfo,
+		setNonRelationshipObjectFieldsInfo,
+	] = useState<nonRelationshipObjectFieldsInfo[]>();
+
+	const {errors, handleChange, setValues, values} = useObjectDetailsForm({
+		initialValues: {
+			defaultLanguageId: selectedNode.data?.defaultLanguageId,
+			externalReferenceCode: selectedNode.data?.externalReferenceCode,
+			id: selectedNode.data?.id,
+			label: {},
+			name: selectedNode.data?.name,
+			pluralLabel: {},
+		},
```

Comment:
Hey @igor-franca, I suggest that you don't put values ​​of a selectedNode in the initial values, the reason for that is because if we choose in the LeftSidebar an object from other folders, we will break the screen because we don't have that object in the Diagram as a Node. Initial values ​​are intended to not break the screen until the request completes or receive stable values ​​from the backend. So you can pass more generic values here like:

```ts
initialValues: {
	defaultLanguageId: 'en_US',
	externalReferenceCode: '',
	id: 0,
	label: {},
	name: '',
	pluralLabel: {},
},
```

Even with that we still have a problem selecting Items in LeftSidebar from other folders, I think we only have to open the RightSidebarObjectDefinitionDetails if the selected definition is from current folder opened in Model Builder, otherwise I think we have to change the bar type side to empty state. What do you think?

Reply:
> Hey @igor-franca, I suggest that you don't put values ​​of a selectedNode in the initial values, the reason for that is because if we choose in the LeftSidebar an object from other folders, we will break the screen because we don't have that object in the Diagram as a Node. Initial values ​​are intended to not break the screen until the request completes or receive stable values ​​from the backend. So you can pass more generic values here like:
> 
> ```ts
> initialValues: {
> 	defaultLanguageId: 'en_US',
> 	externalReferenceCode: '',
> 	id: 0,
> 	label: {},
> 	name: '',
> 	pluralLabel: {},
> },
> ```
> 
> Even with that we still have a problem selecting Items in LeftSidebar from other folders, I think we only have to open the RightSidebarObjectDefinitionDetails if the selected definition is from current folder opened in Model Builder, otherwise I think we have to change the bar type side to empty state. What do you think?

I agree, it makes sense

Reply:
On second thought, I believe we shouldn't use the `SET_SELECTED_NODE` function for definitions that aren't in the current folder.